### PR TITLE
[cuda] [test] Add TI_DEVICE_MEMORY_GB and TI_DEVICE_MEMORY_FRACTION environment variable

### PR DESCRIPTION
**Pull request by archibate**
Related issue = #765

Sol. 1 in https://github.com/taichi-dev/taichi/issues/765#issuecomment-612942650

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

### CODE REVIEW

**Comment by archibate** [comment]
There is still some crashes, however, should I turn to `1 / (threads + 3)`?

**Comment by archibate** [comment]
`fatal: unable to access 'https://github.com/taichi-dev/taichi.git/': Could not resolve host: github.com`
Seems CI fail due to github.com was suffering a ddos attack?

**Comment by archibate** [comment]
Hello? Travis failed is due to mysterious reasons. I think we had better merge this before getting conflicts...

**Comment by yuanming-hu** [comment]
I'll take care of this later today since I have too many meetings during the day. Sorry!

File: cmake/TaichiCore.cmake [hunk 1]
```diff
@@ -77,7 +77,7 @@ endif()
 
 # http://llvm.org/docs/CMake.html#embedding-llvm-in-your-project
 find_package(LLVM REQUIRED CONFIG)
-message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
+message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION} in ${LLVM_DIR}")
```
Comment (yuanming-hu) [inline cmake/TaichiCore.cmake:80]: Again, could you please revert this? It doesn't belong to this PR.

File: python/taichi/main.py [hunk 2]
```diff
@@ -41,6 +41,7 @@ def test_python(args):
             threads = min(8, cpu_count())  # To prevent running out of memory
         except:
             threads = 2
+        os.environ['TI_DEVICE_MEMORY_FRACTION'] = str(1.0 / (threads + 1))
```
Comment (yuanming-hu) [inline python/taichi/main.py:44]: Maybe setting `TI_DEVICE_MEMORY_GB` to something like 250MB leads to more consistent behavior on different machines? (I guess all tests can run successfully with 250GB of memory, but please confirm on your end.)

Comment (archibate) [reply python/taichi/main.py:44]: Sadly, test mem usually goes up to 400MB...either run on single thread, or we impl some memory schedule method (i.e. run 100MB + 400MB at same time, then run 200MB + 200MB next time)

**Comment by archibate** [comment]
Found unexpected error other than oom:
```
tests/python/test_ad_basics.py::test_atan2_f64 Running test on arch=Arch.cuda
arch=Arch.cuda default_fp=DataType.float64
[E 04/22/20 09:55:23.427] [llvm_context.cpp:operator()@49] LLVM Fatal Error: Cannot select: intrinsic %llvm.nvvm.atomic.load.add.f64
```

File: python/taichi/lang/__init__.py [hunk 3]
```diff
@@ -121,19 +121,29 @@ def init(arch=None,
     for k, v in kwargs.items():
         setattr(ti.cfg, k, v)
 
-    def boolean_config(key, name=None):
-        if name is None:
-            name = 'TI_' + key.upper()
+    def boolean_config(key):
```
Comment (yuanming-hu) [inline python/taichi/lang/__init__.py:124]: It seems to me that this function is never used. Please remove. Again, please proof-read your code 3-5 times before requesting a review to avoid abuse of the community resources.

**Comment by archibate** [comment]
> proof-read your code 3-5 times

Read-proof!
